### PR TITLE
Auto-update more versions

### PR DESCRIPTION
**Pull request by trask**
Let me know if you'd like to change anything. After you merge this, it should pick up this lagging `javaVersion` that we haven't been updating as a good test that it's working (I've also tested it in my fork).

https://github.com/open-telemetry/opentelemetry.io/blob/0f45ace76c31107d48e767a37447c7ca58097b1e/content/en/docs/instrumentation/java/manual.md?plain=1#L9

### CODE REVIEW

**Review by chalin — COMMENTED** [review]
Oh! I'm so looking forward to reviewing this more closely, but it might not be until later today or tomorrow. If another maintainer approves before then, I'm ok if this lands sooner than later.

I do have some comments about making the script runnable locally, in particular under macOS. I'd made improvements to `.github/workflows/scripts/check-collector-version.sh` so that it could be run locally _and_ in a "dry-run" mode. I'd like to preserve that ability. I don't mind submitting followup changes.

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [inline .github/workflows/scripts/auto-update-version.sh:23]: It would be helpful if the body could contain, at a minimum, a link to the repo's releases; or even better, a list to the specific version. Or both :)

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [inline .github/workflows/scripts/auto-update-version.sh:14]: It would be beneficial if this failed in the case that no match is found. It would help detect when a version field is (possibly) inadvertently rename or (re)moved from one of the target files. WDYT @trask?

**Comment by chalin** [comment]
Just sharing a few ideas of the top. Will share more later.

**Comment by svrnm** [comment]
I like that! 

@chalin I am wondering if we should move those versions into a data/ file and then use from there at the different places, wdyt? (Does not have to be part of this PR)

File: .github/workflows/auto-update-versions.yml
Comment (chalin) [inline .github/workflows/auto-update-versions.yml:25]: Following #2352, we can drop the second file:

```suggestion
          ./.github/workflows/scripts/auto-update-version.sh "opentelemetry-java" "javaVersion" "content/en/docs/instrumentation/java/_index.md"
```

File: .github/workflows/auto-update-versions.yml
Comment (chalin) [inline .github/workflows/auto-update-versions.yml:24]: Why the leading `./` on these lines?

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [inline .github/workflows/scripts/auto-update-version.sh:31]: I'd drop opentelemetrybot, because I want this to be executable locally if/when we might feel the need for an immediate update. If you prefer having a branch of the form `foo/bar` then maybe it can be `auto-update/...`. Otherwise a simple branch seems fine, WDYT?

```suggestion
branch="auto-update-$repo-$latest_version"
```

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [inline .github/workflows/scripts/auto-update-version.sh:27]:
```suggestion
    echo "PR(s) for version $latest_version of $repo already exist:"
    gh pr list --state all --search "in:title $message"
    echo "So we won't create another. Exiting."
```

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [inline .github/workflows/scripts/auto-update-version.sh:37]:
```suggestion
echo "Submitting auto-update PR '$message'."
```

File: .github/workflows/auto-update-versions.yml
Comment (chalin) [reply .github/workflows/auto-update-versions.yml:25]: ⚠️ If `auto-update-version.sh` runs twice, won't the second run inherit the branch from the first run?

Comment (chalin) [reply .github/workflows/auto-update-versions.yml:25]: If so, the script can save the branch when it starts and change back to that branch when it's done.

File: .github/workflows/auto-update-collector-version.yml
Comment (chalin) [inline .github/workflows/auto-update-collector-version.yml:27]: Do any of these env vars need to be set in the new workflow too?

File: .github/workflows/auto-update-collector-version.yml
Comment (chalin) [inline .github/workflows/auto-update-collector-version.yml:21]: If we're no longer using `GITHUB_TOKEN` (vs. `OPENTELEMETRYBOT_GITHUB_TOKEN ` in the new workflow), then `GITHUB_TOKEN` should be deleted from the settings, right?

**Review by chalin — CHANGES_REQUESTED** [review]
Great improvement. See inline comments and questions, thanks.

File: .github/workflows/scripts/auto-update-version.sh
Comment (trask) [reply .github/workflows/scripts/auto-update-version.sh:31]: > if/when we might feel the need for an immediate update

this workflow supports `workflow_dispatch` (see top), which means you can trigger it manually in GitHub Actions any time you want (this is also a really handy way to test the workflow on your fork, push to `main` on your fork, and then run manually trigger the workflow)

File: .github/workflows/auto-update-collector-version.yml
Comment (trask) [reply .github/workflows/auto-update-collector-version.yml:21]: `GITHUB_TOKEN` is a built-in env var (not one that you add in your settings)

File: .github/workflows/auto-update-collector-version.yml
Comment (trask) [reply .github/workflows/auto-update-collector-version.yml:27]: no, these are handled by

```
          git config user.name opentelemetrybot
          git config user.email 107717825+opentelemetrybot@users.noreply.github.com
```

File: .github/workflows/auto-update-versions.yml
Comment (trask) [reply .github/workflows/auto-update-versions.yml]: removed

File: .github/workflows/auto-update-versions.yml
Comment (trask) [reply .github/workflows/auto-update-versions.yml]: fixed by resetting between each update

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [reply .github/workflows/scripts/auto-update-version.sh:33]: Got it, thanks!

File: .github/workflows/scripts/auto-update-version.sh
Comment (trask) [reply .github/workflows/scripts/auto-update-version.sh]: done

File: .github/workflows/auto-update-versions.yml
Comment (chalin) [inline .github/workflows/auto-update-versions.yml:27]: We probably don't need this second reset since the workflow is done after that, right?

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [reply .github/workflows/scripts/auto-update-version.sh]: Is this a change you'll be adding in this or a followup PR @trask (or maybe have someone else do it)?

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [reply .github/workflows/scripts/auto-update-version.sh:14]: Is this a change you'll be adding in this or a followup PR @trask (or maybe have someone else do it)?

File: .github/workflows/auto-update-versions.yml
Comment (trask) [reply .github/workflows/auto-update-versions.yml:27]: ya, just trying to help the next person who adds another line here 😄

File: .github/workflows/scripts/auto-update-version.sh
Comment (trask) [reply .github/workflows/scripts/auto-update-version.sh:14]: ya, I'll add it in this PR

File: .github/workflows/auto-update-versions.yml
Comment (trask) [reply .github/workflows/auto-update-versions.yml:27]:
```suggestion
```

Comment (chalin) [reply .github/workflows/auto-update-versions.yml:27]: It's your call, but it is currently "dead code" so I'd drop it.

File: .github/workflows/scripts/auto-update-version.sh
Comment (chalin) [reply .github/workflows/scripts/auto-update-version.sh]: Oh, sorry, I missed that. Thanks!

File: .github/workflows/scripts/auto-update-version.sh
Comment (trask) [reply .github/workflows/scripts/auto-update-version.sh:18]: done and tested 👍

File: .github/workflows/auto-update-versions.yml
Comment (trask) [reply .github/workflows/auto-update-versions.yml:27]: ^ leaving this suggestion in case you want to commit it no problem

**Comment by trask** [comment]
> I do have some comments about making the script runnable locally, in particular under macOS. I'd made improvements to `.github/workflows/scripts/check-collector-version.sh` so that it could be run locally _and_ in a "dry-run" mode. I'd like to preserve that ability. I don't mind submitting followup changes.

ya, sorry about that, I generally just test jobs in my fork, but I totally support adding those features back

File: .github/workflows/scripts/auto-update-version.sh
Comment (trask) [inline .github/workflows/scripts/auto-update-version.sh:5]: this could be simplified if there is only a single file per repo version

File: .github/workflows/auto-update-collector-version.yml
Comment (chalin) [inline .github/workflows/auto-update-collector-version.yml:11]: I think that we should have a guard like this in the new workflow. At least the `if: github.repository_owner == 'open-telemetry'` part.

**Review by chalin — APPROVED** [review]
Left an inline comment about adding a guard. Otherwise, this LGTM. Thanks!

File: .github/workflows/auto-update-collector-version.yml
Comment (trask) [reply .github/workflows/auto-update-collector-version.yml:11]: this kind of guard used to be helpful, but github disables jobs on (new) forks now, so I haven't been adding them anymore (and you have to remove the guard if you do want to test it on your fork). or is your concern different?

**Review by cartermp — APPROVED** [review]
Approving on account of the question about the guard being addressed

File: .github/workflows/auto-update-collector-version.yml
Comment (chalin) [reply .github/workflows/auto-update-collector-version.yml:11]: Ok, that's good enough for me. Thanks!

**Pull request merged**
